news/7729.bugfix

```diff
@@ -0,0 +1,3 @@
+Cache the result of parse_links() to avoid re-tokenizing a find-links page multiple times over a pip run.
+
+This change significantly improves resolve performance when --find-links points to a very large html page.
```

Comment:
I’m wondering from reading this text whether this should be a `.feature` instead 🤔

Reply:
True, it's also not really a bug at all to avoid caching here -- the caching is just something that pex v1 happened to do (possibly without realizing it), which then was really exposed as a feature (as you've mentioned) when pex v2 started using pip.

Reply:
I have renamed it to `7729.feature` pending further discussion!